url354ce1a2d1_dagaren_gladiator_chess_tgz-pJ8-es_dagaren_gladiator_main_MainJ8/src/es/dagaren/gladiator/representation/BitboardPosition.java

```diff
@@ -0,0 +1,2907 @@
+   {
+      //Se obtiene el desplazamiento para obtener la fila correspondiente a la casilla
+      int offset = rankSquare[squareIndex] << 3;
+      //Se obtiene la fila desplazando el bitboard de ocupadas
+      int occupation = (int)((bbOccupation >>> offset) & 0xFF);
+      //Se devuelve el bitboard de atacadas fila para la casilla y ocupacion
+      return bbRankAttack[squareIndex][occupation];
+   } //Fin de getAtacadasRank(int casilla)
+   
+   /**
+    * 
+    * @param squareIndex
+    * @return
+    */
+   public long getAttackedInFile(int squareIndex)
+   {
+      //Se obtiene el desplazamiento para obtener la columna correspondiente a la casilla
+      int offset = rankSquare[squareRotated90DLeft[squareIndex]] << 3;
+      //Se obtiene la fila (columna rotada) desplazando el bitboard de ocupadasR90
+      int occupation = (int)((bbOccupationR90L >>> offset) & 0xFF);
+      //Se devuelve el bitboard de atacadas columna para la casilla y ocupacion
+      return bbFileAttack[squareIndex][occupation];
+   } //Fin de getAtacadasFile(int casilla)
+
+   /**
+    * 
+    * @param squareIndex
+    * @return
+    */
+   public long getAttackedInMainDiagonal(int squareIndex)
+   {
+      //Se obtiene el desplazamiento para obtener la diagonal principal correspondiente a la casilla
+      int offset = rankSquare[squareRotated45DLeft[squareIndex]] <<3;
+      //Se obtiene la fila (diagonal rotada) desplazando el bitboard de ocupadasR45R
+      int occupation = (int)((bbOccupationR45R >>> offset) & 0xFF);
+      //Se devuelve el bitboard de atacadas diagonal A1H8 para la casilla y ocupacion
+      return bbPrincipalDiagonalAttack[squareIndex][occupation];
+   } //Fin de getAtacadasDiagonalA1H8(int casilla)
+   
+   /**
+    * 
+    * @param squareIndex
+    * @return
+    */
+   public long getAttackedInSecondaryDiagonal(int squareIndex)
+   {
+      //Se obtiene el desplazamiento para obtener la diagonal secndaria correspondiente a la casilla
+      int offset = rankSquare[inverseSquareRotated45DLeft[squareIndex]] <<3;
+      //Se obtiene la fila (diagonal rotada) desplazando el bitboard de ocupadasR45L
+      int occupation = (int) ((bbOccupationR45L >>> offset) & 0xFF);
+      //Se devuelve el bitboard de atacadas diagonal A8H1 para la casilla y ocupacion
+      return bbSecondaryDiagonalAttack[squareIndex][occupation];
+   } //Fin de getAtacadasDiagonalA8H1
+   
+   
+   protected void resetMoves()
+   {
+      movesGenerated = false;
+      movesList = null;
+   }
+   
+   
+   /************* MÉTODOS HEREDADOS DE POSICION ABSTRACTA ********************/
+
+   public boolean loadFen(String posicionFen)
+   {
+      resetMoves();
+      
+      boolean result = super.loadFen(posicionFen);
+      
+      updateOcupied();
+      updateAttackedFromFull();
+      
+      return result;
+   }
+   
+   /* (non-Javadoc)
+    * @see es.dagaren.chesi.Posicion#deshacerMovement()
+    */
+   public boolean undoMove(Movement mov)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 15. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 1486-1492 into a separate method.